### PR TITLE
Link sketches

### PR DESCRIPTION
**Pull request by matthiaskern**
Opening this PR to keep myself accountable.
Adding a link doesn't yet work on the first opened sketch, opening another sketch fixes it.
Any idea why this could happen @thangngoc89?

TODO still:
- [x] cypress tests
- [x] Fix above bug
- [ ] improve UI design
- [x] signal link load/error status
- [x] ~~implement update link functionality~~ (show timestamp)
- [x] link/execute api should be similar
- [x] add validation when trying to add a module with same id/name
- [x] ~~deal with links that itself have links~~ (i'd propose ignoring those for now)

Future PR should add a cache on jsoo side for the links.

### CODE REVIEW

**Comment by thangngoc89** [comment]
Wow. This is fast :) 

> Adding a link doesn't yet work on the first opened sketch, opening another sketch fixes it.

I'll look into this but I don't know what to produce this. Could you add an cypress test case ? 

Also, while the changes to Editor_Blocks are small, could you please rebase with master? I refactored that monster today. It's quite readable now

**Comment by matthiaskern** [comment]
@thangngoc89 I'm pretty sure it has to do with state not being passed correctly. I'll rebase later, that might help already :)

File: client/src_editor/Editor_Types.re [hunk 1]
```diff
@@ -98,3 +103,19 @@ module Block = {
...
+module Link = {
+  type internalLink = {
+    revision_at: Js.Json.t,
+    sketch_id: string,
+    name: string,
+    lang,
+    code: string,
+  };
+
+  type externalLink = unit;
```
Comment (thangngoc89) [inline client/src_editor/Editor_Types.re]: should be an abstract type `type externalLink;` to make sure this can't be initiated

**Comment by thangngoc89** [comment]
@matthiaskern How is it going with this PR ? Do you think it's ready for merging? (aside from design perspective?)

**Comment by matthiaskern** [comment]
@thangngoc89 Just merged master back in to this. I think this is good to go, but two more small details that I could add tomorrow: 
- we need to explain that a link with name foo is available as module Foo more clearly
- module id's should be rendered as links

**Comment by thangngoc89** [comment]
@matthiaskern Thank you. I'll work on improving the UI of this feature.

**Pull request merged**